File: server/src/main/java/com/fluffy/submission/ui/request/SubmissionWebRequest.java
package com.fluffy.submission.ui.request;

import com.fluffy.global.web.Accessor;
import com.fluffy.submission.application.request.QuestionResponseRequest;
import com.fluffy.submission.application.request.SubmissionRequest;
import java.util.List;

public record SubmissionWebRequest(List<QuestionResponseRequest> questionResponses) {

    public SubmissionRequest toAppRequest(Long examId, Accessor accessor) {
        return new SubmissionRequest(examId, questionResponses, accessor);
    }
}

File: server/src/main/kotlin/com/fluffy/submission/ui/request/SubmissionWebRequest.kt
package com.fluffy.submission.ui.request

import com.fluffy.global.web.Accessor
import com.fluffy.submission.application.request.QuestionResponseRequest
import com.fluffy.submission.application.request.SubmissionRequest

data class SubmissionWebRequest(
    val questionResponses: List<QuestionResponseRequest>
) {

    fun toAppRequest(examId: Long, accessor: Accessor): SubmissionRequest {
        return SubmissionRequest(
            examId = examId,
            questionResponses = questionResponses,
            accessor = accessor
        )
    }
}